File: src/main/java/com/tms/csp/ast/XorCount.java
package com.tms.csp.ast;

public class XorCount implements Comparable<XorCount> {

    private final Exp xor;
    private final String prefix;
    private Integer count = 0;

    public XorCount(Exp xor) {
        this.xor = xor;
        prefix = xor.getPrefix();
    }

    public void increment() {
        count++;
    }

    public int getCount() {
        return count;
    }

    @Override
    public int compareTo(XorCount that) {
        return this.count.compareTo(that.count);
    }

    public Exp getXor() {
        return xor;
    }

    public String getPrefix() {
        return prefix;
    }

    @Override
    public String toString() {
        return prefix + ":" + count;
    }
}

File: src/main/java/com/smartsoft/csp/ast/XorCount.kt
package com.smartsoft.csp.ast

class XorCount(val xor: Xor) : Comparable<XorCount> {

    val prefix: String = xor.prefix

    var count: Int = 0
        private set

    fun increment(amount: Int = 1) {
        count += amount
    }


    override fun compareTo(that: XorCount): Int {
        return this.count.compareTo(that.count)
    }

    override fun toString(): String {
        return "$prefix:$count"
    }
}
